### ISSUE TITLE
FT600 support

### ISSUE THREAD

**Issue by kbeckmann** [lambdaconcept/ft60x_driver#1]
Just wanted to say thanks a lot for writing this module first!

It is a bit hard-coded for the ft601 however it seems to work with ft600 by changing a bit here and there. Would you like an outsider like me try to make it work with both or do you prefer to fix it yourself? I sadly don't have any ft601 hardware to test with, only ft600 currently.

**Comment by key2** [comment]
I'll have to re-write it because it has a design issue. Will do asap. but I don't have ft600 so I can't tell what the differences are. What are you using the ft600 with ?

**Comment by kbeckmann** [comment]
I'm currently bringing up my FPGA board Kilsyth which is basically just an FT600 + ECP5 + SDRAM and a bunch of IO pins. I'm currently bringing up the board, just soldered the first unit. (Twitter thread if you are curious https://twitter.com/kbeckmann/status/1071915370536165377 )

**Comment by kbeckmann** [comment]
The biggest difference between ft600 and 601 is that ft600 has a 16 bit wide data interface and that BE is 2 bit wide instead of 4. The datasheet specifies that ft600 can be used with 4 channel mode but I wonder if that's just a typo/copy paste error.

> When the FT600 or FT601 is configured as 4 channels in the multi-Channel FIFO bus mode, the FIFO
buffer is configured as 1 KB * 2 (double buffered) each on RX and TX channel.

I haven't worked with the FT60x before so this is all new to me, so I don't have a lot of useful feedback to give to you.

**Comment by key2** [comment]
So that confirms that both chip are the same, ft600 is probably the same
die, just less wires bonded...
Are you using it in 2 channel 8 bits mode, or 1 channel 16 bits ?

**Comment by kbeckmann** [comment]
> ft600 is probably the same die

Yeah I believe so too, they are very similar according to the datasheet at least.

> Are you using it in 2 channel 8 bits mode, or 1 channel 16 bits ?

I'll probably use 1 channel 16 bits to start with.

**Comment by Doralitze** [comment]
I'm a bit confused here, as this discussion is 6 years old and has not been resolved. Are FT600 devices supported by now? If not, I'm probably going to use one in the coming weeks and could help debugging, if desired.

**Comment by jeanthom** [comment]
The driver only supports FT601 at the moment. There has been some rewriting effort a few years ago: https://github.com/lambdaconcept/ft60x_driver/tree/rewrite. I can't comment on its usability and what devices it supports.

If you don't mind having a userland driver, https://github.com/apertus-open-source-cinema/ft60x-rs advertises FT600 support (as yet untested).
